### PR TITLE
Implement text color change

### PR DESCRIPTION
**Pull request by weeman1337**
- New prop `textColor`
- `ColorPicker` is now a little more abstract
- Specialised `ElementColorPicker` and `TextColorPicker` exist

text-color

**:heavy_check_mark: Checklist**

-  [x] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [x] Tests for new functionality and regression tests for bug fixes.
- [x] Screenshots or videos attached (for UI changes).
- [x] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

**Comment by nils-nord** [comment]
Test failed.

The AC says, for multi-select it should be the color of the first item selected, instead it always is the color of the text first created from the texts selected. See attachment.
https://github.com/nordeck/matrix-neoboard/assets/97087047/a783839e-83da-4b02-91a4-f74a7ce54c07

File: src/components/ElementBar/ColorPickerButton/ColorPicker.tsx [hunk 1]
```diff
@@ -16,34 +16,37 @@
...
+  calculateUpdatesFn: (elements: Elements, color: string) => ElementUpdate[];
+  label: string;
+};
 
-  const { activeColor, setActiveColor } = useLayoutState();
+export function ColorPicker({
+  color,
+  setColor,
+  calculateUpdatesFn,
+  Icon,
```
Comment (mgcm) [inline src/components/ElementBar/ColorPickerButton/ColorPicker.tsx:43]: is there a reason why the `Icon` and `calculateUpdatesFn` order is different here?

File: src/components/ElementBar/ColorPickerButton/ColorPicker.tsx [hunk 2]
```diff
@@ -16,34 +16,37 @@
...
 import { ColorsGrid } from './ColorsGrid';
-import { calculateColorChangeUpdates } from './calculateColorChangeUpdates';
-import { extractFirstNonTransparentOrFirstColor } from './extractFirstNonTransparentOrFirstColor';
 
-export function ColorPicker() {
-  const { t } = useTranslation();
+export type ColorPickerProps = {
+  color?: string;
+  setColor?: (color: string) => void;
+  /** The icon component to be used for the colour picker in the element bar */
```
Comment (mgcm) [inline src/components/ElementBar/ColorPickerButton/ColorPicker.tsx:32]: Currently, most comments use `colour` and all code occurrences use `color`. This is obviously not an issue but would be good to have consistency here.

Being picky about color pickers is sort of a pun :)

**Review by mgcm — COMMENTED** [review]
Left a couple of comments but this looks good to merge.

File: src/components/ElementBar/ColorPickerButton/ColorPicker.tsx (hunk 1 above)
Comment (weeman1337) [reply src/components/ElementBar/ColorPickerButton/ColorPicker.tsx:43]: Compared to where? It is an object, so order does not matter…

**Pull request merged**